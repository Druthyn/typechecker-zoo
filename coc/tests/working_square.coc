inductive Nat : Type with
  | zero : Nat
  | succ : Nat -> Nat

axiom mult : Nat -> Nat -> Nat

def square (x : Nat) : Nat :=
  mult x x