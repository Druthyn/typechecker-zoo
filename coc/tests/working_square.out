Parsed module with 3 declarations
âœ“ Module 'tests/working_square.coc' type checks successfully!